src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -743,16 +743,34 @@ Section*
 XclBin::findSection(enum axlf_section_kind _eKind,
                     const std::string& _indexName) const
 {
-  for (unsigned int index = 0; index < m_sections.size(); ++index) {
-    if (m_sections[index]->getSectionKind() == _eKind) {
-      if (m_sections[index]->getSectionIndexName().compare(_indexName) == 0) {
-        return m_sections[index];
+  for (auto& section : m_sections) {
+    if (section->getSectionKind() == _eKind) {
+      if (section->getSectionIndexName().compare(_indexName) == 0) {
+        return section;
       }
     }
   }
   return nullptr;
```

Comment:
Please use early-exit to reduce scopes.

```suggestion
  for (auto& section : m_sections) {
    if (section->getSectionKind() != _eKind) 
      continue;
      
    if (section->getSectionIndexName().compare(_indexName) == 0)
      return section;
  }
  return nullptr;
```

Reply:
If we start doing this, why not just fusing the 2 `if` conditions to just have a single `if` since there are a lot of places where there are perfectly nested `if` without `else`?

Reply:
Sure works for me.  But sometimes code reads better with shorter lines.  Makes it more clear that you are looking for a specific section before testing it.  I don't really care.

Reply:
Yes, sorry for nit-picking. :-) But if short lines are better, `section->getSectionIndexName().compare(_indexName) == 0` can perhaps be shortened to some less obfuscated `section->getSectionIndexName() == _indexName`?